### ISSUE TITLE
Types: WrapPointer arguement is not always a number. 

### ISSUE THREAD

**Issue by DennisSmolek** [jrouwe/JoltPhysics.js#126]
When setting up listeners I need to wrap pointers in the callbacks. For example from the contactListener example:
```js
body1 = Jolt.wrapPointer(body1, Jolt.Body);
body2 = Jolt.wrapPointer(body2, Jolt.Body);
```
In this example `body1` and `body2` are `Jolt.BodyID`'s but the `wrapPointer` is set as a number :
`Types.d.ts Line 7`
```js
    function wrapPointer<C extends new (...args: any) => any>(ptr: number, Class: C): InstanceType<C>;
```
does it need to be an `unknown` like `castObject`?
```js
    function castObject<C extends new (...args: any) => any>(object: unknown, Class: C): InstanceType<C>;
```

Typescript is complaining:
```
Argument of type 'BodyID' is not assignable to parameter of type 'number'.ts(2345)
```

**Comment by isaac-mason** [comment]
Seems the root issue here is JSImplementation method arguments don't have correct types.

Currently the ContactListenerJS type is:

```ts
    class ContactListenerJS {
        constructor();
        OnContactValidate(inBody1: Body, inBody2: Body, inBaseOffset: RVec3, inCollisionResult: CollideShapeResult): number;
        OnContactAdded(inBody1: Body, inBody2: Body, inManifold: ContactManifold, ioSettings: ContactSettings): void;
        OnContactPersisted(inBody1: Body, inBody2: Body, inManifold: ContactManifold, ioSettings: ContactSettings): void;
        OnContactRemoved(inSubShapePair: SubShapeIDPair): void;
    }
```

But the webidl binder calls the methods with pointers, so this needs to change to:

```ts
    class ContactListenerJS {
        constructor();
        OnContactValidate(inBody1: number, inBody2: number, inBaseOffset: number, inCollisionResult: number): number;
        OnContactAdded(inBody1: number, inBody2: number, inManifold: number, ioSettings: number): void;
        OnContactPersisted(inBody1: number, inBody2: number, inManifold: number, ioSettings: number): void;
        OnContactRemoved(inSubShapePair: number): void;
    }
```

I'll cut a fix in `webidl-dts-gen` shortly.

**Comment by isaac-mason** [comment]
I've released a fix for webidl-dts-gen under 1.9.0: https://github.com/pmndrs/webidl-dts-gen/releases/tag/webidl-dts-gen%401.9.0

And raised a PR to bump it's version here: https://github.com/jrouwe/JoltPhysics.js/pull/130

**Comment by jrouwe** [comment]
Thanks for fixing this! I've released a new package 0.20.0 with this fix.

**Comment by isaac-mason** [comment]
No worries!
